packages/ymax-planner/src/resolver.ts

```diff
@@ -0,0 +1,75 @@
+import { type SigningSmartWalletKit } from '@agoric/client-utils';
+import type { OfferSpec } from '@agoric/smart-wallet/src/offers';
+import type { CctpSubscription, GmpSubscription,  } from './subscription-manager.js';
+
+type ResolveCCTPSubscriptionParams = {
+  signingSmartWalletKit: SigningSmartWalletKit;
+  subscriptionId: string;
+  status: 'success' | 'timeout';
+  subscriptionData: CctpSubscription;
+  previousOfferId: string;
+  proposal?: object;
+};
+
+type ResolveGMPSubscriptionParams = {
+  signingSmartWalletKit: SigningSmartWalletKit;
+  subscriptionId: string;
+  status: 'success' | 'timeout';
+  subscriptionData: GmpSubscription;
+  previousOfferId: string;
+  proposal?: object;
+};
+
+export const resolveCCTPSubscription = async ({
```

Comment:
Nit: I'm not a fan of XMLHttpRequest-style capitalization. What would you think of treating all these "CCTP" initialisms as if they were complete words?
```suggestion
export const resolveCctpSubscription = async ({
```